firefox_puppeteer/ui/toolbars.py

```diff
@@ -0,0 +1,337 @@
+        elif evt == 'shortcut':
+            cmd_key = self.l10n.get_localized_entity(LocationBar.dtds,
+                                                     'reloadCmd.commandkey')
+            self.urlbar.send_keys(cmd_key)
+        elif evt == 'shortcut2':
+            self.urlbar.send_keys(self.keys.F5)
+
+    def focus(self, evt='click'):
+        """Focus the location bar according to the provided event.
+
+        :param evt: The event to synthesize in order to focus the urlbar
+                    (one of 'click' or 'shortcut').
+        """
+        if evt == 'click':
+            self.urlbar.click()
+        elif evt == 'shortcut':
+            cmd_key = self.l10n.get_localized_entity(LocationBar.dtds,
+                                                     'openCmd.commandkey')
+            (self.marionette.find_element(By.ID, 'main-window')
+                            .send_keys(self.keys.ACCEL, cmd_key))
+        else:
+            raise ValueError("An unknown event type was passed: %s" % evt)
+
+        Wait(self.marionette).until(
+            lambda _: self.urlbar.get_attribute('focused') == 'true')
+
+
+class AutocompleteResults(BaseLib):
+    """Library for interacting with autocomplete results.
+    """
+
+    def __init__(self, *args, **kwargs):
+        BaseLib.__init__(self, *args, **kwargs)
+        # TODO: A "utility" module that sets up the client directly would be
+        # useful here.
+        self.l10n = L10n(self.get_marionette)
+        self.keys = Keys(self.get_marionette)
+
+    @property
+    def popup(self):
+        """Returns a list of auto complete result elements.
+        """
+        return self.marionette.find_element(By.ID,
+                                            'PopupAutoCompleteRichResult')
+
+    @property
+    def results(self):
+        """Returns the container node for autocomplete results.
+        """
+        return self.popup.find_element('anon attribute',
+                                       {'anonid': 'richlistbox'})
+
+    def is_open(self):
+        return self.popup.get_attribute('state') == 'open'
+
+    def get_visible_results(self):
+        """Returns a list of visible result nodes."""
+        return self.marionette.execute_script("""
+          let rv = [];
+          let node = arguments[0];
+          let count = node.itemCount;
+          for (let i = 0; i < count; ++i) {
+            let item = node.getItemAtIndex(i);
+            if (!item.hasAttribute("collapsed")) {
+              rv.push(item);
+            }
+          }
+          return rv;
+        """, script_args=[self.results])
+
+    def close(self, force=False):
+        if self.is_open():
+            if force:
+                self.marionette.execute_script("""
+                  arguments[0].hidePopup();
+                """, script_args=[self.popup])
+            else:
+                (self.marionette.find_element('id', 'urlbar')
+                                .send_keys(Keys.ESCAPE))
```

Comment:
Once we have the reference to a browser window, this could be come `browser.navbar.locationbar.send_keys`